tests/integration-tests/src/test/java/org/apache/activemq/artemis/tests/integration/amqp/AmqpNoHearbeatsTest.java

```diff
@@ -80,4 +88,68 @@ public void inspectOpenedResource(Connection connection) {
       connection.close();
    }
 
+   private static final String QUEUE_NAME = "queue://testHeartless";
+
+   // This test is validating a scenario where the client will leave with connection reset
+   // This is done by setting soLinger=0 on the socket, which will make the system to issue a connection.reset instead of sending a
+   // disconnect.
+   @Test(timeout = 60000)
+   public void testCloseConsumerOnConnectionReset() throws Exception {
```

Comment:
I dont really understand why this test is in the AmqpNoHearbeatsTest class, it feels like it should operate about the same regardless whether AMQP idle-timeout handling is enabled or not. Its definitely not where I would go looking for such a test.

It also seems far more appropriate to testing the changes made in #2139 for ARTEMIS-1927 (without test).

Reply:
The two are related... you should be able to cleanup consumer, disable heartbeats and call disconnect on windows. ARTEMIS-1927 was found after I disabled heart beats and killed a consumer on windows which issued a connection reset.

Reply:
To me it seems that the broker behaviour should essentially be the same here regardless whether idle-timeout is enabled or not, i.e a connection explicitly goes away, the broker cleans up a consumer on it, and its not really testing idle-timeout behaviour or the lack of it, making it feels a bit odd it being in here.